### PR TITLE
feat: can pass --auth flag to `fedimint-cli dev api`

### PR DESCRIPTION
**Pull request by justinmoon**
I couldn't figure out how to make authenticated RPC requests using `fedimint-cli dev api` and formatting the `params` argument correctly. Tried many variations on `'[{"auth":"pass","params":null}]'` and gave up!

Now you can invoke like:

```
fedimint-cli dev api --peer-id 0 --auth pass audit
```

### CODE REVIEW

**Comment by justinmoon** [comment]
Well after going through all this trouble to run the `audit` RPC command via CLI, just realized there is `fedimint-cli  --our-id 0 --password pass admin audit`!

**Comment by justinmoon** [comment]
I think fedimint-cli has top-level `OUR_ID` and auth args. Perhaps `fedimint-cli dev api` should just use those?

File: fedimint-cli/src/lib.rs [hunk 1]
```diff
@@ -395,6 +395,8 @@ Examples:
         /// Which server to send request to
         #[clap(long = "peer-id")]
         peer_id: Option<u16>,
+        #[clap(long = "auth")]
```
Comment (dpc) [inline fedimint-cli/src/lib.rs:398]: Missing comment

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.2`:
- #3786

**Comment by justinmoon** [comment]
Thanks for finishing this off @elsirion

**Pull request merged**